### PR TITLE
Strict STM package

### PR DESCRIPTION
**Pull request by coot**
- strict-stm package
- clean cabal files

### CODE REVIEW

File: strict-stm/NOTICE [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Copyright 2019 Input Output (Hong Kong) Ltd.
```
Comment (dnadales) [inline strict-stm/NOTICE:1]: - 2022?

**Review by bolt12 — APPROVED** [review]
LGTM

File: strict-stm/strict-stm.cabal [hunk 2]
```diff
@@ -0,0 +1,66 @@
+name:                strict-stm
+version:             0.1.0.0
+synopsis:            Type classes for concurrency with STM, ST and timing
+-- description:
+license:             Apache-2.0
+license-files:
+  LICENSE
+  NOTICE
+copyright:           2019-2021 Input Output (Hong Kong) Ltd.
```
Comment (dnadales) [inline strict-stm/strict-stm.cabal:9]: Would it make sense to bump the year already?

File: strict-stm/strict-stm.cabal [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                       -Wpartial-fields
+                       -Widentities
+
+  if flag(asserts)
+    ghc-options: -fno-ignore-asserts
+
+  if flag(checktvarinvariant)
+    cpp-options: -DCHECK_TVAR_INVARIANT
+
+-- test-suite test
```
Comment (dnadales) [inline strict-stm/strict-stm.cabal:55]: Do we want to remove this?

**Review by dnadales — APPROVED** [review]

**Comment by coot** [comment]
bors merge

**Comment by coot** [comment]
bors cancel

File: strict-stm/strict-stm.cabal (hunk 2 above)
Comment (coot) [reply strict-stm/strict-stm.cabal:9]: I thought about it, but decided to bump all of them at once.

File: strict-stm/strict-stm.cabal (hunk 3 above)
Comment (coot) [reply strict-stm/strict-stm.cabal:55]: I don't think so, don't you use the invariant somewhere in `ouroboros-cosnensus`?  Even if not, I think it's still useful.

**Comment by coot** [comment]
bors merge

File: strict-stm/strict-stm.cabal (hunk 3 above)
Comment (dnadales) [reply strict-stm/strict-stm.cabal:55]: Sorry, I meant the commented code :)

Comment (coot) [reply strict-stm/strict-stm.cabal:55]: Oh, yes it doesn't make sense here.

**Pull request merged**